src/core/hle/result.h

```diff
@@ -402,17 +402,16 @@ constexpr bool EvaluateResultFailure(const Result& r) {
 }
 
 template <typename T>
-constexpr void UpdateCurrentResultReference(T result_reference, Result result) {
-    ASSERT(false);
-}
+constexpr void UpdateCurrentResultReference(T result_reference, Result result);
```

Comment:
Is this not able to be explicitly deleted like:

```suggestion
constexpr void UpdateCurrentResultReference(T result_reference, Result result) = delete;
```

To cause a compiler error if invoked at compile-time, but allows the overloads of it to still function?